### PR TITLE
[ci] add license check tool

### LINKED ISSUE #3: [ci] add license check tool
Refer to the `flink-shaded` project to use the `flink-ci-tools` tool to check the license, because the detection notice file in `flink-ci-tools` must contain the content of `flink`, we cannot use this tool class.

The paimon project needs to create a tool to check license.

### PR DESCRIPTION
**Pull request by zhuangchong**
this pr closes #3 

Add paimon-ci-tools source code refer to flink-ci-tools module in flink

### CODE REVIEW

File: paimon-shade-caffeine-2/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -1,5 +1,5 @@
 paimon-shade-caffeine-2
-Copyright 2023-2023 The Apache Software Foundation
+Copyright 2014-2023 The Apache Software Foundation
```
Comment (JingsongLi) [inline paimon-shade-caffeine-2/src/main/resources/META-INF/NOTICE]: Flink Table Store was developed in 2022, I think here should be 2022-2023.

**Review by JingsongLi — COMMENTED** [review]
Thanks @zhuangchong for the contribution, can you add the source (from Flink xxx module) in PR comment?

File: paimon-shade-caffeine-2/src/main/resources/META-INF/NOTICE (hunk 1 above)
Comment (JingsongLi) [reply paimon-shade-caffeine-2/src/main/resources/META-INF/NOTICE]: Sorry, start with `Apache Paimon(incubating)`.
Here should be 2023-2023.

Comment (JingsongLi) [reply paimon-shade-caffeine-2/src/main/resources/META-INF/NOTICE]: We should modify all copyright to 2023-2023.

Comment (zhuangchong) [reply paimon-shade-caffeine-2/src/main/resources/META-INF/NOTICE]: done.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**